tasks/preload.js

```diff
@@ -136,7 +136,14 @@ module.exports = function (grunt) {
 					iMinor = parseInt(aVersionMatch[2], 10);
 				}
 
-				if (options.compatVersion === "edge" || (iMajor === 1 && iMinor >= 40) || iMajor > 1) {
+				if (options.compatVersion === "edge" || (iMajor === 1 && iMinor >= 54) || iMajor > 1) {
+					// Build library-preload as .js file
+					preloadInfo.ext = ".js";
+					preloadInfo.processContent = function(content, preloadObject) {
+						// workaround for new modules based on sap.ui.predefine
+						return 'sap.ui.require.preload(' + JSON.stringify(preloadObject.modules, null, '\t') + ', "' + preloadObject.name + '");';
+					};
+				} else if (iMajor === 1 && iMinor >= 40 && iMinor < 54) {
```

Comment:
The < minor version 54 is not really required but makes it easier to understand the check and would be safe when moving later